### DISCUSSION TITLE
OSRAM RT RGBW (new device) - UnhandledPromiseRejectionWarning

### DISCUSSION THREAD

**Discussion by ndrwstn** [Koenkk/zigbee2mqtt#9685] [Devices]
I have OSRAM LIGHTIFY RT RGBW (recessed lights) which are (to my knowledge) largely identical to Sylvania 73741 which are supported by zigbee2mqtt. I was trying to follow the 'new device' instructions by copying the Sylvania config, but I get a `(node:8) UnhandledPromiseRejectionWarning: SyntaxError: Unexpected token '-'`?

My file.js below:
```
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const fz = {zigbee-herdsman-converters.require('zigbee-herdsman-converters/converters/fromZigbee'), legacy: require('zigbee-herdsman-converters/lib/legacy').fromZigbee};
const ota = require('zigbee-herdsman-converters/lib/ota');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const e = exposes.presets;

module.exports = [
    {
    zigbeeModel: ['RT RGBW'],
    model: 'RT RGBW',
    vendor: 'OSRAM',
    description: 'LIGHTIFY RT RGBW',
    extend: extend.ledvance.light_onoff_brightness_colortemp_color(),
    ota: ota.ledvance,
    },
];
```
Does anyone have any suggestions or what information is needed to troubleshoot this issue? Thanks very much for any help.

**Comment by Koenkk** [comment]
The syntax on line 2 is incorrect, use https://github.com/Koenkk/zigbee2mqtt.io/blob/master/docs/externalConvertersExample/light.js as an example instead

**Reply by ndrwstn** [reply]
Thanks very much.

I'm able to get the lights to work using that example, though with a few problems (LQI = null, for example). If I try to use  `extend.ledvance.light_onoff_brightness_colortemp_color(),` it fails (I haven't tried to see if `ota.ledvance` works on its own). I think that if I can figure out how to pull over the ledvance settings from the other devices, they should work correctly.

If you can point me in the right direction as to how to do that, I'd greatly appreciate it.

Edit to add: I'm also getting weird `Assertion failed` in my logs without any surrounding context (logs are set to debug). I'm also getting what seems to be regular crashes?

```
Error: Unrecognized nodeId:11549
    at Driver.networkIdToEUI64 (/app/node_modules/zigbee-herdsman/src/adapter/ezsp/driver/driver.ts:454:19)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at EZSPAdapter.processMessage (/app/node_modules/zigbee-herdsman/src/adapter/ezsp/adapter/ezspAdapter.ts:50:33)
Using '/data' as data directory
```

**Reply by Koenkk** [reply]
- `extend.ledvance` adds some ledvance specific commands, so no big difference
- coudl you provide the color temp range than I can add this device:  https://www.zigbee2mqtt.io/advanced/support-new-devices/01_support_new_devices.html#_3-adding-converter-s-for-your-device
- Assert Failed error seems to be an error in the ezsp driver (which is experimental); please report it here: https://github.com/Koenkk/zigbee-herdsman/issues/319

**Reply by ndrwstn** [reply]
I think I was wrong that the OSRAMs and Sylvanias are the same, because the command from [the guide](https://www.zigbee2mqtt.io/advanced/support-new-devices/01_support_new_devices.html#_3-adding-converter-s-for-your-device)  returns this error:

```
Zigbee2MQTT:debug 2021-11-16 18:24:13: Received MQTT message on 'zigbee2mqtt/recessed_dressing_1/set' with data '{"read": {"cluster": "lightingColorCtrl", "attributes": ["colorTempPhysicalMin", "colorTempPhysicalMax"]}}'
Zigbee2MQTT:debug 2021-11-16 18:24:13: Publishing 'set' 'read' to 'recessed_dressing_1'
Zigbee2MQTT:error 2021-11-16 18:24:33: Publish 'set' 'read' to 'recessed_dressing_1' failed: 'Error: Read 0x8418260000085d65/3 lightingColorCtrl(["colorTempPhysicalMin","colorTempPhysicalMax"], {"sendWhenActive":false,"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Timeout - 21394 - 3 - 11 - 768 - 1 after 10000ms)'
Zigbee2MQTT:debug 2021-11-16 18:24:33: Error: Read 0x8418260000085d65/3 lightingColorCtrl(["colorTempPhysicalMin","colorTempPhysicalMax"], {"sendWhenActive":false,"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Timeout - 21394 - 3 - 11 - 768 - 1 after 10000ms)
    at Timeout._onTimeout (/app/node_modules/zigbee-herdsman/src/utils/waitress.ts:64:35)
    at listOnTimeout (internal/timers.js:557:17)
    at processTimers (internal/timers.js:500:7)
Zigbee2MQTT:info  2021-11-16 18:24:33: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"Publish 'set' 'read' to 'recessed_dressing_1' failed: 'Error: Read 0x8418260000085d65/3 lightingColorCtrl([\"colorTempPhysicalMin\",\"colorTempPhysicalMax\"], {\"sendWhenActive\":false,\"timeout\":10000,\"disableResponse\":false,\"disableRecovery\":false,\"disableDefaultResponse\":true,\"direction\":0,\"srcEndpoint\":null,\"reservedBits\":0,\"manufacturerCode\":null,\"transactionSequenceNumber\":null,\"writeUndiv\":false}) failed (Timeout - 21394 - 3 - 11 - 768 - 1 after 10000ms)'","meta":{"friendly_name":"recessed_dressing_1"},"type":"zigbee_publish_error"}'
```

I tried it on multiple lights and it fails on the OSRAMs but succeeded for the Sylvanias. Should I just work through the [fromZigbee converters](https://github.com/Koenkk/zigbee-herdsman-converters/blob/master/converters/fromZigbee.js) and try to blueprint these lights manually? In the alternative, these lights are spec'd at 1900-6500K (which should be 154-526 mireds), though I can't imagine it'd be that easy.

Thanks for your ongoing help.

**Reply by Koenkk** [reply]
I forgot to check this first, but this device is indeed already supported: https://github.com/Koenkk/zigbee-herdsman-converters/blob/ab8aaf431ed1f11d24a49fac6b383db2e5e0b920/devices/sylvania.js#L52 (so no need to add support for it anymore, the vendor/model is just a cosmetic thingy)

**Reply by ndrwstn** [reply]
That's great. How do I tell it to use that definition, then? I'm running `koenkk/zigbee2mqtt:1.22.0` and because of the cosmetic differences, they showed up as unsupported.

**Reply by Koenkk** [reply]
The matching is only done on the `zigbeeModel`, which if I understood correctly is the same (`RT RGBW`), could you share the data/database.db entry of your device?

**Reply by ndrwstn** [reply]
That's my mistake. My example was wrong and when I re-created it using your link I apparently corrected it unknowingly. The zigbeeModel is `LIGHTIFY RT RGBW`. One of the relevant entries in my `database.db` is:

```
{"id":52,"type":"Router","ieeeAddr":"0x841826000006a185","nwkAddr":11549,"manufId":48042,"manufName":"OSRAM","powerSource":"Mains (single phase)","modelId":"LIGHTIFY RT RGBW","epList":[3],"endpoints":{"3":{"profId":260,"epId":3,"devId":258,"inClusterList":[0,3,4,5,6,8,768,64527],"outClusterList":[25],"clusters":{"genBasic":{"attributes":{"modelId":"LIGHTIFY RT RGBW","manufacturerName":"OSRAM","powerSource":1,"zclVersion":1,"appVersion":1,"stackVersion":2,"hwVersion":1,"dateCode":"20160215N****"}},"lightingColorCtrl":{"attributes":{}}},"binds":[],"configuredReportings":[],"meta":{}}},"appVersion":1,"stackVersion":2,"hwVersion":1,"dateCode":"20160215N****","zclVersion":1,"interviewCompleted":true,"meta":{"configured":2},"lastSeen":1637162903399}
```

In case it is relevant, I've that light seemingly working (without using ledvance or ota) with the following extension:

```
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const ota = require('zigbee-herdsman-converters/lib/ota');
const e = exposes.presets;
const ea = exposes.access;

module.exports = [
    {
    zigbeeModel: ['LIGHTIFY RT RGBW'],
    model: 'RT RGBW',
    vendor: 'OSRAM',
    description: 'LIGHTIFY RT RGBW 5/6',
    extend: extend.light_onoff_brightness_colortemp_color(),
    // ota: ota.ledvance,
    },
];
```

I apologize for my mistake and thanks for bearing with me here.

**Reply by Koenkk** [reply]
No problem, if you could provide the color temp range + a link (URL) to this device I can add it.https://www.zigbee2mqtt.io/advanced/support-new-devices/01_support_new_devices.html#_3-adding-converter-s-for-your-device

**Reply by ndrwstn** [reply]
You might have missed [my earlier reply](https://github.com/Koenkk/zigbee2mqtt/discussions/9685#discussioncomment-1653445), but the command in that guide did not work for me:

```
Zigbee2MQTT:error 2021-11-18 12:59:10: Publish 'set' 'read' to 'recessed_living_b1' failed: 'Error: Read 0x8418260000084b6a/3 lightingColorCtrl(["colorTempPhysicalMin","colorTempPhysicalMax"], {"sendWhenActive":false,"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Status 'UNSUPPORTED_ATTRIBUTE')'
Zigbee2MQTT:info  2021-11-18 12:59:10: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"Publish 'set' 'read' to 'recessed_living_b1' failed: 'Error: Read 0x8418260000084b6a/3 lightingColorCtrl([\"colorTempPhysicalMin\",\"colorTempPhysicalMax\"], {\"sendWhenActive\":false,\"timeout\":10000,\"disableResponse\":false,\"disableRecovery\":false,\"disableDefaultResponse\":true,\"direction\":0,\"srcEndpoint\":null,\"reservedBits\":0,\"manufacturerCode\":null,\"transactionSequenceNumber\":null,\"writeUndiv\":false}) failed (Status 'UNSUPPORTED_ATTRIBUTE')'","meta":{"friendly_name":"recessed_living_b1"},"type":"zigbee_publish_error"}'
```

Do I need to remove my custom [external converter](https://github.com/Koenkk/zigbee2mqtt/discussions/9685#discussioncomment-1658949) or should it still work? Thanks again.

**Reply by Koenkk** [reply]
No problem could you provide a link (URL) of this device?

**Reply by ndrwstn** [reply]
[PDF](https://d2a7t4o9z5arn6.cloudfront.net/GoodMart/pdfs/sylvania/LEDlightifyRT5RT6.pdf) is the best information I can find. They are 1900k-6500k.

**Reply by Koenkk** [reply]
Thanks, added the device.

Changes will be available in the dev branch in a few hours from now. (https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html)
